test/unit/metrics/test_metricframe_bootstrap_values.py

```diff
@@ -0,0 +1,180 @@
+            np.full(n_B_incorrect, "B"),
+        )
+    )
+
+    # Make sure we have avoided rounding errors
+    assert len(s_f) == n_samples
+
+    # Create the metric frame
+    # The ci argument is set to have the
+    # nominal value, as well as the one and two
+    # sigma values
+    mf = MetricFrame(
+        metrics=accuracy_score,
+        y_true=y_true,
+        y_pred=y_pred,
+        sensitive_features=s_f,
+        n_boot=10000,
+        ci=(
+            0.025,
+            0.1587,
+            0.5,
+            0.8413,
+            0.975,
+        ),  # -2sigma, -sigma, nominal, sigma, 2sigma
+    )
+
+    return mf
+
+
+class TestBootstrapValues:
+    def test_overall(self, mf):
+        assert mf.overall == p
+
+    def test_by_group(self, mf):
+        assert mf.by_group["A"] == p_A
+        assert mf.by_group["B"] == p_B
+
+    def test_overall_ci(self, mf):
+        mean = p * n_samples
+        sigma = math.sqrt(n_samples * p * (1 - p))
+
+        # Following corresponds to the ci argument to MetricFrame
+        sigma_vals = [-2, -1, 0, 1, 2]
+
+        for i in range(len(sigma_vals)):
+            value = mean + (sigma_vals[i] * sigma)
+            value_frac = value / n_samples
+
+            assert mf.overall_ci[i][1] == pytest.approx(
+                value_frac, rel=1 / math.sqrt(n_samples)
+            )
+
+    def test_by_group_ci(self, mf):
+        mean_A = p_A * n_A
+        sigma_A = math.sqrt(n_A * p_A * (1 - p_A))
+
+        mean_B = p_B * n_B
+        sigma_B = math.sqrt(n_B * p_B * (1 - p_B))
+
+        # Following corresponds to the ci argument to MetricFrame
+        sigma_vals = [-2, -1, 0, 1, 2]
+
+        for i in range(len(sigma_vals)):
+            v_A = mean_A + (sigma_vals[i] * sigma_A)
+            vf_A = v_A / n_A
+
+            v_B = mean_B + (sigma_vals[i] * sigma_B)
+            vf_B = v_B / n_B
+
+            # Slightly wider tolerance here.... not sure if this should be required
+            assert mf.by_group_ci[i][1]["A"] == pytest.approx(
+                vf_A, rel=2 / math.sqrt(n_A)
+            )
+            assert mf.by_group_ci[i][1]["B"] == pytest.approx(
+                vf_B, rel=2 / math.sqrt(n_B)
+            )
+
+    def test_difference_between_groups(self, mf):
+        # When adding or subtracting, absolute errors
+        # add in quadrature
+        # Only check the one sigma values here
+
+        # Start by checking the nominal value
+        assert mf.difference(method="between_groups") == pytest.approx(
+            mf.difference_group_ci[2][1], rel=1 / math.sqrt(n_samples)
+        )
+
+        # Do the error calculation
+        nominal_acc_AB = abs(p_A - p_B)
+
+        sigma_A = math.sqrt(n_A * p_A * (1 - p_A))
+        sigma_B = math.sqrt(n_B * p_B * (1 - p_B))
+        acc_sigma_A = sigma_A / n_A
+        acc_sigma_B = sigma_B / n_B
+
+        # The actual quadrature
+        acc_sigma_AB = math.sqrt(acc_sigma_A**2 + acc_sigma_B**2)
```

Comment:
Again @MiroDudik  and @harsha-nori, I'd appreciate it if you double check my work here. It's been a while since I did error calculations.

Reply:
It sounds like you're suggesting testing the guts of `create_ci_output()` in `_bootstrap.py` - the stuff around `np.quantile()` specifically. I would actually be happy to do that; this was an option because I understood what was supposed to be going on at this high level. The expected output of `create_ci_output()` was not so easy to discern...